intro.md

```diff
@@ -1,9 +1,21 @@
 # Introduction
 
 ## Design patterns
-
-What are design patterns? What are idioms? Anti-patterns.
+When developing programs, we have to solve many problems. A program can be viewed as a solution to a problem. It can also be viewed as a collection of solutions to many different problems. All of these solutions work together to solve a bigger problem.
 
 ## Design patterns in Rust
 
-Why Rust is a bit special - functional elements, type system - borrow checker
+There are many problems that share the same form. Due to the fact that Rust is not object-oriented design patterns vary with respect to other object-oriented programming languages. While the details are different, since they have the same form they can be solved using the same fundamental methods. 
```

Comment:
Not sure why the review comments go away, but I've added a sentence here in reference to what you've recommended

Reply:
yes!